src/zarr/abc/store.py

```diff
@@ -172,6 +172,16 @@ async def set(self, key: str, value: Buffer) -> None:
         """
         ...
 
+    @abstractmethod
+    async def setdefault(self, key: str, default: Buffer) -> None:
```

Comment:
After merging in `main`, there's a failing test in the new logging store. Adding this method there should get things green again...

```python
    async def setdefault(self, key: str, value: Buffer) -> None:
        with self.log():
            return await self._store.setdefault(key=key, value=value)
```